### PR TITLE
CHORE: Prepare native base to remove

### LINKED ISSUE #33: Remove native-base
Right now we are using native base and maintainers of the native base are abandoning the support for this library.

That means we need to find other alternative and use it in the template.

### PR DESCRIPTION
**Pull request by MateuszRostkowski**
# Description

The first stage of migrating the native-base to another library OR custom solution (we haven't decided yet).
It provides updates on imports to make the migration as painless as possible.

Fixes # ([issue-33](https://github.com/binarapps/expo-ts-template/issues/33))

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Verify if UI looks similar to the previous on updated custom components.
- [x] Verify if App don't crash on imports update.

# Screenshot(s)

## Android
Screenshot_1693914926
Screenshot_1693914948
Screenshot_1693914965
Screenshot_1693914971

## iOS
Simulator Screenshot - iPhone 14 Pro - 2023-09-05 at 13 50 05
Simulator Screenshot - iPhone 14 Pro - 2023-09-05 at 13 50 15
Simulator Screenshot - iPhone 14 Pro - 2023-09-05 at 13 50 28
Simulator Screenshot - iPhone 14 Pro - 2023-09-05 at 13 50 38

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] Add correct label to your pull request
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ module.exports = {
     'react/prop-types': 'off',
 
     'react-native/no-single-element-style-arrays': 'off',
+    'react-native/no-color-literals': 'off',
```
Comment (MateuszRostkowski) [inline .eslintrc.js]: We can't add it here, we can't have free colors inside files where react native views are, like `View` or `Text`

File: src/components/atoms/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+      return 'black'
+    }, [disabled, value])
+
+    const bgColor = useMemo(() => {
+      if (!value) {
+        return 'white'
+      }
+      if (disabled) {
+        return '#EBEBE4'
```
Comment (MateuszRostkowski) [inline src/components/atoms/Checkbox.tsx]: This color should come from theme, like gray100 or something.

Comment (MateuszRostkowski) [reply src/components/atoms/Checkbox.tsx]: Use values from useTheme.

File: src/components/atoms/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import { forwardRef, useCallback, useMemo } from 'react'
```
Comment (MateuszRostkowski) [inline src/components/atoms/Checkbox.tsx]: @vercia 
Did you check how this component look on both themes (light/dark)?

File: src/components/atoms/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      }
+      if (disabled) {
+        return '#EBEBE4'
+      }
+
+      return 'transparent'
+    }, [disabled, value])
+
+    const borderColor = useMemo(
+      () => (isError ? 'red' : disabled ? 'grey' : 'black'),
```
Comment (MateuszRostkowski) [inline src/components/atoms/Checkbox.tsx]: Colors from useTheme

File: src/components/atoms/FormErrorMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { StyleSheet, Text } from 'react-native'
+
+export const FormErrorMessage = ({ errorMessage }: { errorMessage?: string }) => {
+  return errorMessage && <Text style={styles.text}>{errorMessage}</Text>
+}
+
+const styles = StyleSheet.create({
+  text: {
+    color: 'red',
```
Comment (MateuszRostkowski) [inline src/components/atoms/FormErrorMessage.tsx]: This should come from useTheme

File: src/components/atoms/FormLabel.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        </Text>
+      )}
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  requiredText: {
+    color: 'red',
+  },
```
Comment (MateuszRostkowski) [inline src/components/atoms/FormLabel.tsx]: useTheme

File: src/components/atoms/Select.tsx [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+      setValue(newValue)
+      return
+    }
+    if (value?.length < maxSelectedItems) {
+      newValue.push(item.value)
+      setValue(newValue)
+    }
+  }, [closeDropdown, item.value, maxSelectedItems, setValue, value])
+
+  const color = useMemo(() => (disabled && !selected ? 'gray' : 'black'), [disabled, selected])
```
Comment (MateuszRostkowski) [inline src/components/atoms/Select.tsx]: use useTheme color values

File: src/components/atoms/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+          />
+        </Box>
+      </BottomSheetModal>
+    </>
+  )
+}
+
+const styles = StyleSheet.create({
+  errorBorder: {
+    borderColor: 'red',
```
Comment (MateuszRostkowski) [inline src/components/atoms/Select.tsx]: useTheme

File: src/components/atoms/Select.tsx [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+    right: 8,
+  },
+  itemWrapper: {
+    padding: 8,
+  },
+  mainWrapper: {
+    justifyContent: 'center',
+  },
+  normalBorder: {
+    borderColor: 'gray',
```
Comment (MateuszRostkowski) [inline src/components/atoms/Select.tsx]: useTheme

File: src/components/atoms/Select.tsx [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+  },
+  textInput: {
+    alignItems: 'center',
+    backgroundColor: 'white',
```
Comment (MateuszRostkowski) [inline src/components/atoms/Select.tsx]: useTheme

File: src/components/molecules/Field/Radio.tsx [hunk 11]
```diff
@@ -1,24 +1,75 @@
...
+import { FormErrorMessage, FormLabel } from '~components/atoms'
+
+export const Radio = forwardRef<any, FieldRadioProps>(
   (
-    { isInvalid, isRequired, isDisabled, value, radioOptions, name, errorMessage, onChange },
+    { isRequired, value, radioOptions, errorMessage, isError, onChange, label, labelStyle },
     ref
   ) => {
+    const borderColor = useMemo(() => (isError ? 'red' : 'black'), [isError])
+    const bgColor = useCallback((item: string) => (item === value ? 'blue' : 'gray'), [value])
```
Comment (MateuszRostkowski) [inline src/components/molecules/Field/Radio.tsx]: useTheme

File: src/components/molecules/Field/Radio.tsx [hunk 12]
```diff
@@ -1,24 +1,75 @@
...
-export const Radio = forwardRef<typeof NBRadio, FieldRadioProps>(
+import { FormErrorMessage, FormLabel } from '~components/atoms'
+
+export const Radio = forwardRef<any, FieldRadioProps>(
   (
-    { isInvalid, isRequired, isDisabled, value, radioOptions, name, errorMessage, onChange },
+    { isRequired, value, radioOptions, errorMessage, isError, onChange, label, labelStyle },
     ref
   ) => {
+    const borderColor = useMemo(() => (isError ? 'red' : 'black'), [isError])
```
Comment (MateuszRostkowski) [inline src/components/molecules/Field/Radio.tsx]: useTheme

File: src/components/atoms/Checkbox.tsx (hunk 3 above)
Comment (MSzalowski) [reply src/components/atoms/Checkbox.tsx]: # Dark theme
image

# Light theme
image

File: src/components/LanguagePicker.tsx [hunk 13]
```diff
@@ -46,14 +43,14 @@ export const LanguagePicker: React.FC = () => {
...
       return (
         <Pressable {...props}>
-          <HStack>
+          <Row>
             <Text fontSize="xl" pr="2">
               {languages[language].emoji}
             </Text>
             <Animated.View style={[animatedIconStyle, styles.icon]}>
-              <ChevronDownIcon />
+              <Icon size={24} name="arrow-down-line" />
```
Comment (MSzalowski) [inline src/components/LanguagePicker.tsx]: It makes the Icon invisible on a light theme

File: src/components/atoms/FormLabel.tsx [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+  const stylesForRequired =
+    labelStyle &&
+    Object.fromEntries(
+      Object.entries(labelStyle).map(([key, value]) =>
+        key === 'color' ? [key, 'red'] : [key, value]
+      )
+    )
+
+  return (
+    <View style={[styles.wrapper, { ...(label && { marginBottom: 8, marginTop: 4 }) }]}>
```
Comment (MSzalowski) [inline src/components/atoms/FormLabel.tsx]: Those styles should be moved to `StyleSheet`

File: src/components/atoms/Select.tsx [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+import { SelectKey, SelectItemProps, SelectProps } from './types'
+
+import { useColorScheme } from '~contexts'
+// TODO: ISSUE-33 (https://github.com/binarapps/expo-ts-template/issues/33)
+// Remove `useTheme` hook when issue is resolved
+import { useTheme } from '~hooks'
+
+const ITEM_HEIGHT = 56
+
+const bottomSheetContentHeight = Dimensions.get('screen').height / 1.5
```
Comment (MSzalowski) [inline src/components/atoms/Select.tsx]: Both consts should be either uppercased or camel-cased, we shouldn't mix naming conventions for the same type of variables.

File: src/components/LanguagePicker.tsx (hunk 13 above)
Comment (MSzalowski) [reply src/components/LanguagePicker.tsx]: Update:
Icon isn't white by default, without color it is transparent - so not visible at all

image

Comment (MSzalowski) [reply src/components/LanguagePicker.tsx]: I've updated icon (check pr description)

**Review by MSzalowski — APPROVED** [review]

**Comment by MSzalowski** [comment]
@vercia Could you have a look in your spare time? Cause I have made changes to your changes (e.g. I removed `hitslop` on Checkbox as Touchables were overflowing each other)

**Review by vercia — APPROVED** [review]
Thank you for your fixes, I checked this and it looks ok 👍

**Pull request merged**